libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -562,81 +517,38 @@ bool AC_WPNav::update_wpnav()
+// returns true if update_wpnav has been run very recently
+bool AC_WPNav::is_active() const
 {
-    // exit immediately if recalc is not required
-    if (_flags.recalc_wp_leash) {
-        calculate_wp_leash_length();
-    }
+    return (AP_HAL::millis() - _wp_last_update) < 200;
 }
 
-/// calculate_wp_leash_length - calculates horizontal and vertical leash lengths for waypoint controller
-void AC_WPNav::calculate_wp_leash_length()
+// returns target yaw in centi-degrees (used for wp and spline navigation)
+float AC_WPNav::get_yaw() const
 {
-    // length of the unit direction vector in the horizontal
-    float pos_delta_unit_xy = norm(_pos_delta_unit.x, _pos_delta_unit.y);
-    float pos_delta_unit_z = fabsf(_pos_delta_unit.z);
-
-    float speed_z;
-    float leash_z;
-    if (_pos_delta_unit.z >= 0.0f) {
-        speed_z = _pos_control.get_max_speed_up();
-        leash_z = _pos_control.get_leash_up_z();
-    }else{
-        speed_z = fabsf(_pos_control.get_max_speed_down());
-        leash_z = _pos_control.get_leash_down_z();
-    }
-
-    // calculate the maximum acceleration, maximum velocity, and leash length in the direction of travel
-    if(is_zero(pos_delta_unit_z) && is_zero(pos_delta_unit_xy)){
-        _track_accel = 0;
-        _track_speed = 0;
-        _track_leash_length = WPNAV_LEASH_LENGTH_MIN;
-    }else if(is_zero(_pos_delta_unit.z)){
-        _track_accel = _wp_accel_cmss/pos_delta_unit_xy;
-        _track_speed = _pos_control.get_max_speed_xy() / pos_delta_unit_xy;
-        _track_leash_length = _pos_control.get_leash_xy()/pos_delta_unit_xy;
-    }else if(is_zero(pos_delta_unit_xy)){
-        _track_accel = _wp_accel_z_cmss/pos_delta_unit_z;
-        _track_speed = speed_z/pos_delta_unit_z;
-        _track_leash_length = leash_z/pos_delta_unit_z;
-    }else{
-        _track_accel = MIN(_wp_accel_z_cmss/pos_delta_unit_z, _wp_accel_cmss/pos_delta_unit_xy);
-        _track_speed = MIN(speed_z/pos_delta_unit_z, _pos_control.get_max_speed_xy() / pos_delta_unit_xy);
-        _track_leash_length = MIN(leash_z/pos_delta_unit_z, _pos_control.get_leash_xy()/pos_delta_unit_xy);
+    if (_flags.wp_yaw_set) {
+        return _yaw;
+    } else {
+        // if yaw has not been set return attitude controller's current target
+        return _attitude_control.get_att_target_euler_cd().z;
     }
-
-    // calculate slow down distance (the distance from the destination when the target point should begin to slow down)
-    calc_slow_down_distance(_track_speed, _track_accel);
-
-    // set recalc leash flag to false
-    _flags.recalc_wp_leash = false;
 }
 
 // returns target yaw in centi-degrees (used for wp and spline navigation)
-float AC_WPNav::get_yaw() const
+float AC_WPNav::get_yaw_rate() const
 {
     if (_flags.wp_yaw_set) {
-        return _yaw;
+        return _yaw_rate;
     } else {
         // if yaw has not been set return attitude controller's current target
-        return _attitude_control.get_att_target_euler_cd().z;
```

Comment:
I've restored the code so this line is no longer being changed. I'll leave the unnecessary "else" in place as well because we haven't touched the lines around it now.